### DISCUSSION TITLE
Prevent TextBox from changing BackGround color when Selected or Hoverover

### DISCUSSION THREAD

**Discussion by EtherGhost** [AvaloniaUI/Avalonia#6937] [Q&A] [answered]
How can I make a TextBox to stay with the BackGround color I have set, even when mouse is hovering over or when TextBox is selected. 
I'm changing BackGround color based on a bool Property and it should stick to that color even if it's selected etc.

**Comment by ahopper** [comment, marked as answer]
create  a style https://docs.avaloniaui.net/docs/styling and copy the bits that change the background from here https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Themes.Fluent/Controls/TextBox.xaml and tweak to the colors you want

**Reply by EtherGhost** [reply]
Thanks @ahopper! Got it! 
Will dive into it next week. 
I actually solved my other question the same way. My questing about custom error template. I will reply myself there.

**Comment by djrecipe** [comment]
Thanks for the leads here guys. I'm struggling with the strange default styles for hovering and selection of textboxes. Example:

![PdfCloudTester_kccOn2QwVm](https://github.com/user-attachments/assets/e75154f3-a345-46da-b622-8ca6e2afb556)

My readonly `TextBox` is white when not hovered, grey when hovered, and black when selected.

Since I'm a bit lazy: do I have to dive into styles to fix this? Would a `HoverColor` and `SelectedColor` be too much property bloat?

**Comment by djrecipe** [comment]
Okay, I was finally able to wrap my head around all this, and here is the styling to create a plain ole `TextBox` which does NOT change regardless if it is hovered over or focused:

```
<Window.Styles>
    <Style Selector="TextBox.normalStyle">
        <Setter Property="Background" Value="Azure"/>
        <Setter Property="Foreground" Value="Black"/>
        <Style Selector="^:pointerover">
            <Setter Property="Foreground" Value="Black" />

            <Style Selector="^ /template/ Border#PART_BorderElement">
                <Setter Property="BorderBrush" Value="Azure"/>
                <Setter Property="Background" Value="Azure" />
            </Style>

            <Style Selector="^ /template/ TextBlock#PART_Watermark">
                <Setter Property="Foreground" Value="Black" />
            </Style>
        </Style>
        <Style Selector="^:focus">
            <Setter Property="Foreground" Value="Black" />

            <Style Selector="^ /template/ TextBlock#PART_Watermark">
                <Setter Property="Foreground" Value="Black" />
            </Style>

            <Style Selector="^ /template/ Border#PART_BorderElement">
                <Setter Property="Background" Value="Azure"/>
                <Setter Property="BorderBrush" Value="Azure"/>
                <Setter Property="BorderThickness" Value="0" />
            </Style>
        </Style>
    </Style>

</Window.Styles>
```
